File: wip/timming/Prgm5.cpp
#include "time.h"
#include <iostream>

using namespace std;

void run(int topCandidate) {
  int candidate = 2;
  while(candidate <= topCandidate) {
    int trialDivisor = 2;
    int prime = 1;
	    
    bool found = true;
    while(((trialDivisor * trialDivisor) <= candidate) && found == true) {
      if(candidate % trialDivisor == 0) {
	prime = 0;
	found = false;
      }
      else {
	trialDivisor++;
      }
    }
	    
    if(found == true) {
      cout << candidate << endl;
    }
    candidate++;
  }
}

int main() {
  long start = clock();
	
  run(1000000);
  
  cout << "---------------------------" << endl;
  cout << "Time: " << (float)(clock() - start) / CLOCKS_PER_SEC << " second(s)." << endl;	  
}

File: wip/timming/prgm5.pl
my $topCandidate = 1000000;
my $candidate = 2;
while($candidate <= $topCandidate) {
	my $trialDivisor = 2;
	my $prime = 1;
	 
	my $found = 1;
	while((($trialDivisor * $trialDivisor) <= $candidate) && $found == 1) {
		if(($candidate % $trialDivisor) == 0) {
			$prime = 0;
			$found = 0; 
		}
		else {
			$trialDivisor++;
		}
	}
	
	if($found == 1) {
		print "$candidate\n";
	}
	$candidate++;
}